tools/swift-format/Sources/Rules/CommentWhitespace.swift

```diff
@@ -11,5 +11,95 @@ import SwiftSyntax
 ///
 /// - SeeAlso: https://google.github.io/swift#horizontal-whitespace
 public final class CommentWhitespace: SyntaxFormatRule {
+  public override func visit (_ token: TokenSyntax) -> Syntax {
+    var pieces = [TriviaPiece]()
+    var validToken = token
+    var isInvalid = false
+    
+    guard let nextToken = token.nextToken else {
+      // In the case there is a line comment at the end of the file, it ensures
+      // that the line comment has a single space after the `//`.
+      pieces = checksSpacesAfterLineComment(isInvalid: &isInvalid, token: token).reversed()
+      return isInvalid ? token.withLeadingTrivia(Trivia.init(pieces: pieces)) : token
+    }
+    
+    // Ensures the line comment has at least 2 spaces before the `//`.
+    if hasLineComment(trivia: nextToken.leadingTrivia) {
+      let numSpaces = token.trailingTrivia.numberOfSpaces
+      if numSpaces < 2 {
+        isInvalid = true
+        let addSpaces = 2 - numSpaces
+        diagnose(.addSpacesBeforeLineComment(count: addSpaces), on:token)
+        validToken = token.withTrailingTrivia(token.trailingTrivia.appending(.spaces(addSpaces)))
+      }
+    }
+    
+    pieces = checksSpacesAfterLineComment(isInvalid: &isInvalid, token: token).reversed()
+    return isInvalid ? validToken.withLeadingTrivia(Trivia.init(pieces: pieces)) : token
+  }
+  
+  /// Returns a boolean indicating if the given trivia contains a line comment.
+  private func hasLineComment (trivia: Trivia) -> Bool {
+    // When the comment isn't inline with code, it doesn't need to
```

Comment:
I think it'd be better to write this in terms of what you're looking for - maybe "Comments are inline unless the trivia begins with a newline."